lib/identity/client.js

```diff
@@ -300,6 +306,95 @@ class Client extends PartialClient {
     )
     return await validateResponseAsJSON(response)
   }
+
+  /**
+   * revokeRecordWithGroup
+   *
+   * @param {String} secretType
+   * @param {String} secretName
+   * @param {String} userToRevokeShare
+   *
+   * @return {Boolean}
+   */
+  async revokeRecordWithGroup(secretName, secretType, userToRevokeShare) {
+    // Validation
+    if (userToRevokeShare == '') {
+      throw new Error('Username Required')
+    }
+    // Look Up the ClientID for Username
+    let clientMapping = await this.searchRealmIdentitiesByUsername([
+      userToRevokeShare,
+    ])
+    let usernameLowerCase = userToRevokeShare.toLowerCase()
+    let clientID =
+      clientMapping.identities_username_to_client_aliases[usernameLowerCase]
+    if (clientID == null) {
+      return null
+    }
+    // Look up group for our current user sharing to the username
+    // The group name is ordered by first clientID is the client who wrote the secret
+    // and second clientID is the user that we are sharing with
+    // This allows visibility to who owns the secret within a pairing
+    const groupName = `tozny.secret.${this.config.realmName}.${this.storage.config.clientId}.${clientID}.${secretName}.${secretType}`
+    const groupList = await this.storage.listGroups(
+      this.storage.config.clientId,
+      [groupName]
+    )
+    if (groupList.groups.length < 2) {
+      // Group Does not Exists, so record is not shared
+      return true
+    } else {
+      // ushare record with group
+      const currentGroup = groupList.groups[0]
+      let recordType = `tozny.secret.${SECRET_UUID}.${secretType}.${secretName}`
+      let revokedFromGroup = await this.storage.revokeRecordWithGroup(
+        currentGroup.groupID,
+        recordType
+      )
+      return revokedFromGroup
+    }
+  }
+
+  /**
+   * getSecretSharedList
+   *
+   * @param {String} secretType
+   * @param {String} secretName
+ 
+   *
+   * @return {Promise<Record>}
+   */
+  async getSecretSharedList(secretName, secretType) {
+    const groupList = await this.storage.listGroups(
+      this.storage.config.clientId
+    )
+    if (groupList.groups.length < 1) {
+      // If there is only one group, this means its the automatic group, no shared groups
+      return []
+    } else {
+      let groupSharedList = []
+      for (let index = 0; index < groupList.groups.length; index++) {
+        let groupNameSplit = groupList.groups[index].groupName.split('.')
+        if (groupNameSplit.length == 7) {
```

Comment:
nit: maybe a `!=` with a `continue` statement? Will keep the code less nested which can make it easier to read.

Also: If doing a `.map()`, this would become a `.filter()` to remove items which don't match the criteria.

```js
.map(
  g =>  { /*do any needed transformation, maybe return false if it doesn't match the criteria*/ }
)
.filter( g => g ) //identity function will filter out falsey values, removing the items marked as false
```